plugin/pkg/scheduler/factory/factory.go

```diff
@@ -164,22 +165,79 @@ func NewConfigFactory(client clientset.Interface, schedulerName string, hardPodA
 		},
 	)
 
-	// TODO(harryz) need to fill all the handlers here and below for equivalence cache
 	c.pVLister.Store, c.pvPopulator = cache.NewInformer(
 		c.createPersistentVolumeLW(),
 		&v1.PersistentVolume{},
 		0,
-		cache.ResourceEventHandlerFuncs{},
+		cache.ResourceEventHandlerFuncs{
+			// MaxPDVolumeCountPredicate: since it relies on the counts of PV.
+			AddFunc: func(obj interface{}) {
+				invalidPredicates := sets.NewString("MaxPDVolumeCountPredicate")
+				pv := obj.(*v1.PersistentVolume)
+				if pv.Spec.AWSElasticBlockStore != nil {
+					invalidPredicates.Insert("MaxEBSVolumeCount")
+				}
+				if pv.Spec.GCEPersistentDisk != nil {
+					invalidPredicates.Insert("MaxGCEPDVolumeCount")
+				}
+				c.equivalencePodCache.InvalidCachedPredicateItemOfAllNodes(invalidPredicates)
+			},
+			DeleteFunc: func(obj interface{}) {
+				invalidPredicates := sets.NewString("MaxPDVolumeCountPredicate")
+				pv := obj.(*v1.PersistentVolume)
+				if pv.Spec.AWSElasticBlockStore != nil {
+					invalidPredicates.Insert("MaxEBSVolumeCount")
+				}
+				if pv.Spec.GCEPersistentDisk != nil {
+					invalidPredicates.Insert("MaxGCEPDVolumeCount")
+				}
+				c.equivalencePodCache.InvalidCachedPredicateItemOfAllNodes(invalidPredicates)
+			},
+		},
+	)
+
+	c.pVCLister.Indexer, c.pvcPopulator = cache.NewIndexerInformer(
```

Comment:
@kargakis yes, although the scheduler is the last one I'm planning on tackling because it has more challenges than the basic controllers.

@resouer we want to use shared informers any time we have 2 or more distinct places in the code that would both be creating and using an informer, *and* the code does not need to manually manipulate the informer's indexer/store.